sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/rules.scala

```diff
@@ -356,8 +358,28 @@ case class PreprocessTableInsertion(conf: SQLConf) extends Rule[LogicalPlan] {
           s"including ${staticPartCols.size} partition column(s) having constant value(s).")
     }
 
-    val newQuery = DDLPreprocessingUtils.castAndRenameQueryOutput(
```

Comment:
I saw there is another usage of `castAndRenameQueryOutput` , for `CreateTable` case, should it get rid of unsafe casts too?

Reply:
In `CreateTable`, it should be CTAS and the schema should be the same as the input query.